File: src/lib/money.ts
/** Half-up rounding to 2 decimal places (paise). See README rounding policy. */
export function roundInr(value: number): number {
  const scaled = value * 100;
  const rounded = Math.sign(scaled) * Math.round(Math.abs(scaled));
  return rounded / 100;
}

/** Half-up rounding to cents (SPEC-US §4.0). */
export function roundUsd(value: number): number {
  return roundInr(value);
}

/** Half-up rounding to pence (SPEC-UK §4.0). */
export function roundGbp(value: number): number {
  return roundInr(value);
}

export function roundMoney(value: number, locale: "IN" | "US" | "UK"): number {
  if (locale === "US") return roundUsd(value);
  if (locale === "UK") return roundGbp(value);
  return roundInr(value);
}

File: mobile/core/financial-planner-core/src/money.rs
/// Half-up rounding to 2 decimal places (paise). Matches `src/lib/money.ts` `roundInr`.
pub fn round_inr(value: f64) -> f64 {
    let scaled = value * 100.0;
    let rounded = scaled.signum() * scaled.abs().round();
    rounded / 100.0
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn round_inr_matches_js_math_round() {
        // f64 quirk: 1.005 * 100 === 100.499... in both JS and Rust
        assert_eq!(round_inr(1.005), 1.0);
        assert_eq!(round_inr(1.004), 1.0);
        assert_eq!(round_inr(49282.445), 49_282.45);
    }
}
